### PR TITLE
fix(ci): don't try to push docs on master push

### PR DESCRIPTION
**Pull request by Avantol13**
### Improvements
- don't try to push docs on master push

### CODE REVIEW

**Review by paulineribeyre — DISMISSED** [review]

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Docs
+
+on:
+  # DON't run on pushes to master and only on pull requests
+  pull_request:
+    types: [opened, reopened]
```
[the remark this answers is not in the document]
Comment (MaribelleHGomez) [reply .github/workflows/docs.yaml:6]: so would we ever really have a pull request on the master branch? I think I might be confused
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

[the remark this answers is not in the document]
Comment (Avantol13) [reply .github/workflows/docs.yaml]: Okay I think I found a way. there's a `synchronize` option

[the remark this answers is not in the document]
Comment (Avantol13) [reply .github/workflows/docs.yaml]: @MaribelleHGomez yeah the problem was that the way it is now is it allow the docs CI to run on master, which tries to make a commit to that "branch", but b/c we have branch protection on master (e.g. you can't commit directly to it), the docs job failed. The behavior should be that the docs job only runs on PRs and commits to real branches, it shouldn't run on master commits that are merged

[the remark this answers is not in the document]
Comment (paulineribeyre) [reply .github/workflows/docs.yaml]: From [here](https://github.com/orgs/community/discussions/24567#discussioncomment-3244508)
> only triggered when the pull request’s tracking branch is synchronized with the source branch for the pull request, and that happens when the source branch is updated.

I think the `synchronize` event might not be triggered when we first open the PR... so maybe `synchronize` + `opened`? 😬

[the remark this answers is not in the document]
Comment (Avantol13) [reply .github/workflows/docs.yaml]: shoot, yeah you're right. I'll update

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 name: CI
 
 on:
+  # run on pushed commits to master and on new commits on pull requests
   push:
+    branches:
+      - master
```
Comment (paulineribeyre) [inline .github/workflows/ci.yaml:7]: ok last comment! sometimes it's nice to have tests results during development, before creating a PR. but i don't have strong feelings

**Review by paulineribeyre — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (Avantol13) [reply .github/workflows/ci.yaml:7]: or specify any branches, I assume that's the default

**Review by paulineribeyre — DISMISSED** [review]

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**